app/main/views/frameworks.py

```diff
@@ -79,18 +81,37 @@ def get_framework(framework_slug):
 
 @main.route('/frameworks/<framework_slug>', methods=['POST'])
 def update_framework(framework_slug):
+    attribute_whitelist = [
+        'status',
+        'clarificationQuestionsOpen',
+        'frameworkAgreementDetails'
+    ]
+
     updater_json = validate_and_return_updater_request()
     framework = Framework.query.filter(
         Framework.slug == framework_slug
     ).first_or_404()
 
     json_payload = get_json_from_request()
     json_has_required_keys(json_payload, ['frameworks'])
-    json_only_has_required_keys(json_payload['frameworks'], ['status', 'clarificationQuestionsOpen'])
+
+    if not json_payload['frameworks']:
+        abort(400, "Framework update expects a payload")
+
+    invalid_keys = (set(json_payload['frameworks'].keys()) - set(attribute_whitelist))
+    if invalid_keys:
+        abort(400, "Invalid keys for framework update: '{}'".format("', '".join(invalid_keys)))
+
+    if 'frameworkAgreementDetails' in json_payload['frameworks']:
+        validate_framework_agreement_details_data(
+            json_payload['frameworks']['frameworkAgreementDetails'],
+            enforce_required=False
+        )
+
+    for key in json_payload['frameworks']:
+        setattr(framework, inflection.underscore(key), json_payload['frameworks'][key])
```

Comment:
Yeah, I agree about iterating over `attribute_whitelist`. And I'm not sure about the use of `inflection` here, I think doing something like
```python
attribute_whitelist = {
    'clarificationQuestionsOpen': 'clarification_questions_open',
    ...
]
```
while a bit repetitive is less likely to fail in surprising ways.